app/bundles/LeadBundle/Model/CompanyReportData.php

```diff
@@ -0,0 +1,153 @@
+<?php
+
+/*
+ * @copyright   2014 Mautic Contributors. All rights reserved
+ * @author      Mautic
+ *
+ * @link        http://mautic.org
+ *
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace Mautic\LeadBundle\Model;
+
+use Mautic\FormBundle\Entity\Field;
+use Mautic\ReportBundle\Event\ReportGeneratorEvent;
+
+class CompanyReportData
+{
+    /**
+     * @var FieldModel
+     */
+    private $fieldModel;
+
+    public function __construct(FieldModel $fieldModel)
+    {
+        $this->fieldModel = $fieldModel;
+    }
+
+    public function getCompanyData()
+    {
+        $companyColumns = $this->getCompanyColumns();
+        $companyFields  = $this->fieldModel->getEntities([
+            'filter' => [
+                'force' => [
+                    [
+                        'column' => 'f.object',
+                        'expr'   => 'like',
+                        'value'  => 'company',
+                    ],
+                ],
+            ],
+        ]);
+
+        $companyColumns = array_merge($companyColumns, $this->getFieldColumns($companyFields, 'comp.'));
+
+        return $companyColumns;
+    }
+
+    /**
+     * @param ReportGeneratorEvent $event
+     *
+     * @return bool
+     */
+    public function eventHasCompanyColumns(ReportGeneratorEvent $event)
+    {
+        $companyColumns = $this->getCompanyColumns();
+        foreach ($companyColumns as $key => $column) {
+            if ($event->hasColumn($key)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * @return array
+     */
+    private function getCompanyColumns()
+    {
+        return [
+            'comp.id' => [
+                'label' => 'mautic.lead.report.company.company_id',
+                'type'  => 'int',
+                'link'  => 'mautic_company_action',
+            ],
+            'comp.companyname' => [
```

Comment:
companyname through companyindustry should be overwritten by getFieldColumns() and thus not needed here.